src/main/java/uk/gov/companieshouse/api/accounts/service/impl/ApprovalService.java

```diff
@@ -45,27 +49,39 @@
     public ApprovalService(
         ApprovalRepository approvalRepository,
         ApprovalTransformer approvalTransformer,
+        ApprovalValidator approvalValidator,
         SmallFullService smallFullService,
         KeyIdGenerator keyIdGenerator) {
         this.approvalRepository = approvalRepository;
         this.approvalTransformer = approvalTransformer;
+        this.approvalValidator = approvalValidator;
         this.smallFullService = smallFullService;
         this.keyIdGenerator = keyIdGenerator;
     }
 
     @Override
     public ResponseObject<Approval> create(Approval rest, Transaction transaction,
         String companyAccountId, HttpServletRequest request) throws DataException {
+
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("transaction_id", transaction.getId());
+        debugMap.put("company_accounts_id", companyAccountId);
+        Errors errors = approvalValidator.validateApproval(rest, request);
+
+        if (errors.hasErrors()) {
+            DataException dataException = new DataException(
+                "Failed to validate " + ResourceName.APPROVAL.getName());
+            LOGGER.errorRequest(request, dataException, debugMap);
+            return new ResponseObject<>(ResponseStatus.VALIDATION_ERROR, rest, errors);
```

Comment:
this shouldn't include the `rest` parameter as it will return an error instead of it so that argument should be null (or we could create a constructor that takes just the rest object and another that just takes the errors to avoid this going forwards